src/potions.ts

```diff
@@ -348,6 +382,117 @@ function useAsValuable(potion: Potion, embezzlers: number, embezzlersOnly: boole
   return total;
 }
 
+const toothpickPotion = new Potion($item`papier-mâché toothpicks`, {
+  price: (historical: boolean) => {
+    return (
+      4 *
+      (npcPrice($item`soda water`) +
+        (historical && historicalAge($item`Rad Lib`) < 14
+          ? historicalPrice($item`Rad Lib`)
+          : mallPrice($item`Rad Lib`)))
+    );
+  },
+  acquire: (
+    qty: number,
+    item: Item,
+    maxPrice?: number | undefined,
+    throwOnFail?: boolean,
+    maxAggregateCost?: number | undefined,
+    tryRetrievingUntradeable?: boolean
+  ) => {
+    const globs = acquire(
+      4 * qty,
+      $item`Rad Lib`,
+      maxPrice ? maxPrice / 2 : mallPrice($item`Rad Lib`) * 2,
+      throwOnFail ?? false,
+      maxAggregateCost ?? 50000,
+      tryRetrievingUntradeable ?? false
+    );
+    Array(Math.floor(globs / 4))
+      .fill(0)
+      .every(() => create(item));
+    return itemAmount(item);
+  },
+});
+
+const rufusPotion = new Potion($item`closed-circuit pay phone`, {
+  providesDoubleDuration: false,
+  canDouble: false,
+  effect: $effect`Shadow Waters`,
+  duration: 30,
+  price: (historical: boolean) => {
+    const canAcquireItemQuest = rufusTarget() === null;
+    const haveItemQuest = get("rufusQuestType", "") === "items";
+    const haveArtifact = get("rufusQuestType", "") === "artifact" && have(rufusTarget() as Item);
+
+    // We will only buff up if we can complete the item quest
+    if (
+      !(
+        canAcquireItemQuest ||
+        haveItemQuest ||
+        haveArtifact ||
+        have($item`Rufus's shadow lodestone`)
+      )
+    ) {
+      return Infinity;
+    }
+
+    // If we are overdrunk, we will need to be able to grab the NC (with a wineglass)
+    if (
+      myInebriety() > inebrietyLimit() &&
+      (!have($item`Drunkula's wineglass`) || !canEquip($item`Drunkula's wineglass`))
+    ) {
+      return Infinity;
+    }
+
+    // We consider the average price of the shadow items to not get gated behind an expensive one
+    const shadowItems = $items`shadow brick, shadow ice, shadow sinew, shadow glass, shadow stick, shadow skin, shadow flame, shadow fluid, shadow sausage, shadow bread, shadow venom, shadow nectar`;
+    const averagePrice =
+      sum(shadowItems, (it) =>
+        historical && historicalAge(it) < 14 ? historicalPrice(it) : mallPrice(it)
+      ) / shadowItems.length;
+
```

Comment:
`garboAverageValue` exists, and `garboAverageValue(items, true)` should use historical price

Reply:
`garboAverageValue` doesn't support a boolean for historical price unfortunately

Reply:
This should probably be sliced out but that can be future you's problem. Or future me. Or future whoever, it's not hard.